sparrow-py/pytests/session_test.py

```diff
@@ -1,13 +1,100 @@
 """Tests for the Kaskada session."""
 from sparrow_py import Session
 import pyarrow as pa
+import random
+import pandas
+import pytest
+from sparrow_py.session import InvalidSchemaException
 
+table_name = "table1"
+time_column = "time"
+key_column = "key"
+schema = pa.schema([
+    pa.field(time_column, pa.int64(), nullable=False), 
+    pa.field(key_column, pa.int64(), nullable=False)
+])
+
+
+def generate_dataset(num_members, num_records_per_member):
+    member_ids = list(range(0, num_members))
+    records = []
+    for member_id in member_ids:
+        for i in range(0, num_records_per_member):
+            records.append({
+                'time': random.randint(1000000000000000, 9000000000000000), # number of seconds from epoch
+                'key': member_id,
+            })
+    df = pandas.DataFrame(records)
+    return df
+
+def generate_dataset_invalid(num_members, num_records_per_member):
```

Comment:
I wonder if these could be a fixture or a class that generates the dataset.

https://docs.pytest.org/en/6.2.x/fixture.html